frontend/src/pages/TracesExplorer/Filter/Filter.tsx

```diff
@@ -0,0 +1,227 @@
+					}
+					durationKey = keys;
+					return acc;
+				}
+
+				if (attributeName) {
+					if (acc[attributeName as AllTraceFilterKeys]) {
+						const existingValue = acc[attributeName as AllTraceFilterKeys];
+						acc[attributeName as AllTraceFilterKeys] = {
+							values: [...existingValue.values, ...values],
+							keys,
+						};
+					} else {
+						acc[attributeName as AllTraceFilterKeys] = { values, keys };
+					}
+				}
+
+				return acc;
+			}, {} as FilterType);
+
+		const durationFinValue = [];
+		if (!isUndefined(durationValue.min)) {
+			durationFinValue.push(getMs(String(durationValue.min)));
+		} else if (durationValue.max) {
+			durationFinValue.push(getMs('0'));
+		}
+
+		if (!isUndefined(durationValue.max)) {
+			durationFinValue.push(getMs(String(durationValue.max)));
+		}
+
+		return durationKey && durationFinValue.length
+			? {
+					...filterRet,
+					durationNano: {
+						keys: durationKey,
+						values: durationFinValue,
+					},
+			  }
+			: filterRet;
+	}, [currentQuery]);
+
+	useEffect(() => {
+		if (!isEqual(syncSelectedFilters, selectedFilters)) {
+			setSelectedFilters(syncSelectedFilters);
+		}
+	}, [syncSelectedFilters]);
+
+	// eslint-disable-next-line sonarjs/cognitive-complexity
+	const preparePostData = (): TagFilterItem[] => {
+		if (!selectedFilters) {
+			return [];
+		}
+		const items = Object.keys(selectedFilters)?.flatMap((attribute) => {
+			const { keys, values } = selectedFilters[attribute as AllTraceFilterKeys];
+			if ((attribute as AllTraceFilterKeys) === 'durationNano') {
+				if (!values.length) {
+					return [];
+				}
+				const minValue = values[0];
+				const maxValue = values[1];
+
+				const minItems: TagFilterItem = {
+					id: uuid().slice(0, 8),
+					op: '>=',
+					key: keys,
+					value: parseInt(minValue, 10) * 1000000,
+				};
+
+				const maxItems: TagFilterItem = {
+					id: uuid().slice(0, 8),
+					op: '<=',
+					key: keys,
+					value: parseInt(maxValue, 10) * 1000000,
+				};
+
+				return maxValue ? [minItems, maxItems] : [minItems];
+			}
+			return {
+				id: uuid().slice(0, 8),
+				key: keys,
+				op: 'in',
+				value: values,
+			};
+		});
+
+		return items as TagFilterItem[];
+	};
+
+	const handleRun = useCallback(
+		(props?: HandleRunProps): void => {
+			const preparedQuery: Query = {
+				...currentQuery,
+				builder: {
+					...currentQuery.builder,
+					queryData: currentQuery.builder.queryData.map((item) => ({
+						...item,
+						filters: {
+							...item.filters,
+							items: props?.resetAll
+								? []
+								: (unionTagFilterItems(item.filters.items, preparePostData())
+										.map((item) =>
+											item.key?.key === props?.clearByType ? undefined : item,
+										)
+										.filter((i) => i) as TagFilterItem[]),
+						},
+					})),
+				},
+			};
+			redirectWithQueryBuilderData(preparedQuery);
+		},
+		[currentQuery, redirectWithQueryBuilderData, selectedFilters],
+	);
+
+	useEffect(() => {
+		handleRun();
```

Comment:
this is leading to queries being run as soon as you change the tag in QB while typing...

Reply:
fixed, using compositeQuery